tbapy/main.py

```diff
@@ -27,7 +38,18 @@ def _get(self, url):
         :param url: URL string to get data from.
         :return: Requested data in JSON format.
         """
-        return get(self.URL_PRE + url, headers={'X-TBA-Auth-Key': self.auth_key}).json()
+        return get(self.READ_URL_PRE + url, headers={'X-TBA-Auth-Key': self.auth_key}).json()
+
+    def _post(self, url, data):
+        """
+        Helper method: POST data to a given URL on TBA's API.
+
+        :param url: URL string to post data to.
+        :pararm data: Dumped data to post.
+        :return: Requests Response object.
+
+        """
+        return post(self.WRITE_URL_PRE + url % self.auth_event_key, data=data, headers={'X-TBA-Auth-Id': self.auth_id, 'X-TBA-Auth-Sig': md5((self.auth_secret + '/api/trusted/v1/' + url + data).encode('utf-8')).hexdigest()})
```

Comment:
I think you could do json=data instead, which would convert the data to json automatically instead of you having to use json.dumps.

Reply:
The only reason I do json.dumps earlier is because the auth-sig needs it to be in that form.